lib/terraform/runner.rb

```diff
@@ -0,0 +1,193 @@
+      #
+      # @return [Terraform::Runner::Response] Response object with result of terraform run
+      def fetch_result_by_stack_id(stack_id)
+        retrieve_stack_job(stack_id)
+      end
+
+      # =================================================
+      # TerraformRunner Stack-API interaction methods
+      # =================================================
+      private
+
+      def server_url
+        ENV.fetch('TERRAFORM_RUNNER_URL', 'https://opentofu-runner:27000')
+      end
+
+      def server_token
+        ENV.fetch('TERRAFORM_RUNNER_TOKEN', nil)
+      end
+
+      def stack_job_interval_in_secs
+        ENV.fetch('TERRAFORM_RUNNER_STACK_JOB_CHECK_INTERVAL', 10).to_i
+      end
+
+      def stack_job_max_time_in_secs
+        ENV.fetch('TERRAFORM_RUNNER_STACK_JOB_MAX_TIME', 120).to_i
+      end
+
+      # Create to paramaters as used by terraform-runner api
+      #
+      # @param vars [Hash] Hash with key/value pairs that will be passed as input variables to the
+      #        terraform-runner run
+      # @return [Array] Array of {:name,:value}
+      def convert_to_cam_parameters(vars)
+        return [] if vars.nil?
+
+        vars.map do |key, value|
+          {
+            :name  => key,
+            :value => value
+          }
+        end
+      end
+
+      # create http client for terraform-runner rest-api
+      def terraform_runner_client
+        @terraform_runner_client ||= begin
+          # TODO: verify ssl
+          verify_ssl = false
+
+          Faraday.new(
+            :url => server_url,
+            :ssl => {:verify => verify_ssl}
+          ) do |builder|
+            builder.request(:authorization, 'Bearer', -> { server_token })
+          end
+        end
+      end
+
+      def stack_tenant_id
+        '00000000-0000-0000-0000-000000000000'.freeze
+      end
+
+      def json_post_arguments(payload)
+        return JSON.generate(payload), "Content-Type" => "application/json".freeze
+      end
+
+      # Create TerraformRunner Stack Job
+      def create_stack_job(
+        template_path,
+        input_vars: [],
+        tags: nil,
+        credentials: [],
+        env_vars: {},
+        name: "stack-#{rand(36**8).to_s(36)}"
+      )
+        _log.info("start stack_job for template: #{template_path}")
+        tenant_id = stack_tenant_id
+        encoded_zip_file = encoded_zip_from_directory(template_path)
+
+        # TODO: use tags,env_vars
+        payload = {
+          :cloud_providers => credentials,
+          :name            => name,
+          :tenantId        => tenant_id,
+          :templateZipFile => encoded_zip_file,
+          :parameters      => convert_to_cam_parameters(input_vars)
+        }
+        # _log.debug("Payload:>\n, #{payload}")
+
+        http_response = terraform_runner_client.post(
+          "api/stack/create",
+          *json_post_arguments(payload)
+        )
+        _log.debug("==== http_response.body: \n #{http_response.body}")
```

Comment:
Not for this PR, but we have a plugin logger (`$embedded_terraform_log`) which is typically used for api client logging we could probably use that when we're just dumping the request / response to the logs